### PR TITLE
Add logic for automatically searching for extensions when editing files with unsupported extensions

### PR DESCRIPTION
**Pull request by Feel-ix-343**
Initial support for the task "Add logic for automatically searching for extensions when editing files with unsupported extensions" in issue #7096 

Given that language servers have not been offloaded to plugins yet, this is not useful right now, but it surely will be in the future. Also, the backend API does not yet send a *language* field in the extension manifest, so this PR instead searches the extension name and description to find compatible extensions given the editor's set language and the filename. Ideally, we will filter by a language field from the API response, but we cannot yet. When the backend is updated, the implementations of [this trait](https://github.com/zed-industries/zed/compare/main...Feel-ix-343:zed:main#diff-d4fc4b231ff6a74f424c9832e24479e780027c98fca1cd264d77720db591ef88R166) will change accordingly.

Please leave feedback as I am happy to make changes. 

This was my first attempt at working with the codebase, so I have likely used some of the APIs incorrectly. Please let me know!

zedupdates
openfilesnotif

### CODE REVIEW

**Comment by JosephTLyons** [comment]
The one thing I'm curious about here is, how to prevent the toast from showing up on subsequent file opens if you decide you don't want to install a plugin. I'm wondering if we could store some json in Zed's local database about which file types have prompted users to install extensions, and not resurface the toast more than once.

This is just my gut instinct, but I'm sure other members on the team have different opinions - curious to hear your and their thoughts.

**Comment by RemcoSmitsDev** [comment]
It would also be nice to disable all the suggestions, because I think it can be annoying for some people.

File: crates/extensions_ui/Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,8 @@ theme.workspace = true
 ui.workspace = true
 util.workspace = true
 workspace.workspace = true
+language.workspace = true
```
Comment (maxdeviant) [inline crates/extensions_ui/Cargo.toml:30]: Can we sort these in alphabetical order?

File: crates/extensions_ui/src/extension_suggest.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+use crate::ExtensionsWithQuery;
+
+
+/// Function to initialize subscriptions to Editor Open and Buffer Language Change; called in the Extensions UI init function
+pub(crate) fn init(cx: &mut AppContext) {
+
+    // Watch when the editor is opened
+    cx.observe_new_views(move |editor: &mut Editor, cx| {
+
+        // Only check if the editor is editing in a programming language
```
Comment (maxdeviant) [inline crates/extensions_ui/src/extension_suggest.rs:19]: I think we can remove most of these comments, as they're primarily restating what the code is already doing.

Comment (Feel-ix-343) [reply crates/extensions_ui/src/extension_suggest.rs:19]: Ok!

**Comment by Feel-ix-343** [comment]
Great! This is better since I can't test the extension database locally anyway

**Pull request closed without merging**